src/lib.rs

```diff
@@ -85,25 +85,38 @@ impl Bundle {
 
         if let Some(variables) = variables {
             for variable in variables {
-                let key: String = variable.0.to_string();
+                // Make sure the variable key is a Python string,
+                // raising a TypeError if not.
+                let python_key = variable.0;
+                if !python_key.is_instance_of::<PyString>() {
+                    return Err(PyTypeError::new_err(format!(
```

Comment:
Thought (out of scope): I'm beginning to wonder if we might be able to represent this whole for-loop-context with a `match` statement? :thinking:

Reply:
Yeah, there's certainly room for improvement! I'm still a noob at Rust. 😊

Reply:
Me too :D